pkg/controller/replication/conversion.go

```diff
@@ -247,6 +249,11 @@ func (c conversionClient) Patch(ctx context.Context, name string, pt types.Patch
 	return nil, errors.New("Patch() is not implemented for conversionClient")
 }
 
+func (c conversionClient) Apply(context.Context, string, appsav1pply.ReplicaSetBuilder, string, metav1.ApplyOptions, ...string) (result *apps.ReplicaSet, err error) {
```

Comment:
Not implementing Apply would give the user a compile time error making it easier to use.
Instead of returning an error.
But that would require an Applier client interface or sth.

Reply:
For backward compatibility, the Apply function will only be generated if client-gen is called flag is set to opt-in to have it generated. For kubernetes/kubernetes, I've enabled generation of Apply functions and am fixing any issues it causes. There is still the problem for anyone implementing the client interfaces of the kubernetes/kubernetes types, but they're just going to have to resolve those.